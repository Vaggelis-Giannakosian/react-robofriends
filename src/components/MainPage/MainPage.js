import React from 'react'
import Header from "../Header";
import SearchBox from "../SearchBox/SearchBox";
import Scroll from "../Scroll/Scroll";
import ErrorBoundry from "../ErrorBoundry";
import CardList from "../CardList/CardList";

import './MainPage.css'

const MainPage = (props) => {

    return (
        <div className="tc">
            <Header/>
            <SearchBox searchChange={ props.searchChange }/>
            <Scroll>
                <ErrorBoundry>
                    <CardList robots={ props.robots }/>
                </ErrorBoundry>
            </Scroll>
        </div>
    );
}


export default MainPage;