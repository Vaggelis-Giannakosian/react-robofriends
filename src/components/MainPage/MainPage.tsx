import React from 'react'
import Header from "../Header";
import SearchBox, {ISearchCallback} from "../SearchBox/SearchBox";
import Scroll from "../Scroll/Scroll";
import ErrorBoundry from "../ErrorBoundry";
import CardList from "../CardList/CardList";

import './MainPage.css'
import {IRobot} from "../../containers/App/App";

interface IProps {
    searchChange: ISearchCallback,
    robots:  IRobot[]
}


const MainPage = (props : IProps) => {

    return (
        <div className="tc">
            <Header/>
            <SearchBox searchChange={ props.searchChange }/>
            <Scroll>
                <ErrorBoundry>
                    <CardList robots={ props.robots }/>
                </ErrorBoundry>
            </Scroll>
        </div>
    );
}


export default MainPage;
