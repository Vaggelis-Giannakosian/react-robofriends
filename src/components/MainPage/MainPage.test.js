import MainPage from './MainPage'
import {shallow, mount, render} from 'enzyme'
import {robots} from "../../robots";


it('Main Page',()=>{
    const mounted = shallow(<MainPage robots={ robots }/>)

    expect(mounted.length).toEqual(1)
    expect(mounted).toMatchSnapshot()
})