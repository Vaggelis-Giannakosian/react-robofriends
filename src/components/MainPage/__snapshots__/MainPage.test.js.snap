// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main Page 1`] = `
<div
  className="tc"
>
  <Memo(Header) />
  <Memo(SearchBox) />
  <Scroll>
    <ErrorBoundry>
      <CardList
        robots={
          Array [
            Object {
              "email": "Sincere@april.biz",
              "id": 1,
              "name": "Leanne Graham",
              "username": "Bret",
            },
            Object {
              "email": "Shanna@melissa.tv",
              "id": 2,
              "name": "Ervin Howell",
              "username": "Antonette",
            },
            Object {
              "email": "Nathan@yesenia.net",
              "id": 3,
              "name": "Clementine Bauch",
              "username": "Samantha",
            },
            Object {
              "email": "Julianne.OConner@kory.org",
              "id": 4,
              "name": "Patricia Lebsack",
              "username": "Karianne",
            },
            Object {
              "email": "Lucio_Hettinger@annie.ca",
              "id": 5,
              "name": "Chelsey Dietrich",
              "username": "Kamren",
            },
            Object {
              "email": "Karley_Dach@jasper.info",
              "id": 6,
              "name": "Mrs. Dennis Schulist",
              "username": "Leopoldo_Corkery",
            },
            Object {
              "email": "Telly.Hoeger@billy.biz",
              "id": 7,
              "name": "Kurtis Weissnat",
              "username": "Elwyn.Skiles",
            },
            Object {
              "email": "Sherwood@rosamond.me",
              "id": 8,
              "name": "Nicholas Runolfsdottir V",
              "username": "Maxime_Nienow",
            },
            Object {
              "email": "Chaim_McDermott@dana.io",
              "id": 9,
              "name": "Glenna Reichert",
              "username": "Delphine",
            },
            Object {
              "email": "Rey.Padberg@karina.biz",
              "id": 10,
              "name": "Clementina DuBuque",
              "username": "Moriah.Stanton",
            },
          ]
        }
      />
    </ErrorBoundry>
  </Scroll>
</div>
`;
